### PR TITLE
Checkspelloc

### PR DESCRIPTION
**Pull request by jgdumas**
Adds a check of local smith form via sparse elimination modulo a prime power

### CODE REVIEW

File: tests/test-smith-form-local.C [hunk 1]
```diff
@@ -170,240 +153,29 @@ static bool testLocalSmith (const LocalPIR &R, vector<typename LocalPIR::Element
...
 	static Argument args[] = {
-		{ 'm', "-m M", "Set dimension of test matrices to MxN.", TYPE_INT,     &m },
-		{ 'n', "-n N", "Set dimension of test matrices to MxN.", TYPE_INT,     &n },
-		{ 'r', "-r R", "Set rank of test matrices to R.", TYPE_INT,     &r },
-		{ 'i', "-i i", "Set # repetitions.", TYPE_INT,     &iter },
-		{ 'q', "-q Q", "Operate over the ring Z/q^eZ.", TYPE_INTEGER, &q },
-		{ 'e', "-e e", "Operate over the ring Z/q^eZ.", TYPE_INT, &e },
-		{ 'd', "-d d", "Density of random sparse matrices.", TYPE_DOUBLE, &d },
+		{ 'n', "-n N", "Set dimension of test matrices to NxN.", TYPE_INT,     &n },
+		{ 'q', "-q Q", "Operate over the ring Z/q^eZ.", TYPE_INT, &q },
```
Comment (ClementPernet) [inline tests/test-smith-form-local.C]: the doc string still mentions exponent e

**Comment by jgdumas** [comment]
It was like that before the branch, but yes we can add e to the param, see 850b63d.
The test is broken anyway (returns no error, but outputs "ERROR: Computed invariants incorrect").

File: tests/test-smith-form-local.C [hunk 2]
```diff
@@ -155,28 +153,31 @@ static bool testLocalSmith (const LocalPIR &R, vector<typename LocalPIR::Element
...
 		{ 'q', "-q Q", "Operate over the ring Z/q^eZ.", TYPE_INT, &q },
-		//{ 'q', "-q Q", "Operate over the ring Z/qZ.", TYPE_INTEGER, &q },
+        { 'e', "-e e", "Operate over the ring Z/q^eZ.", TYPE_INT, &e },
+        { 's', "-s S", "Random generator seed.", TYPE_INT,     &rseed }	,
 		END_OF_ARGUMENTS
 	};
 
 	parseArguments (argc, argv, args);
+    std::srand(rseed);
+    FFLAS::writeCommandString(std::cout, args) << std::endl;
```
Comment (ClementPernet) [inline tests/test-smith-form-local.C]: This breaks the standard output interface of LinBox test suite (and will cause the CI to fail)

**Comment by jgdumas** [comment]
the standard output interface of LinBox test suite is fine with cout 
(both make check and travis/jenkins will handle output correctly),
whereas for benchmarks it indeed needs to respect some structure.

**Comment by ClementPernet** [comment]
Good to go

**Pull request merged**